app/models/queries/projects/filters/ancestor_filter.rb

```diff
@@ -26,69 +26,63 @@
 # See COPYRIGHT and LICENSE files for more details.
 #++
 
-module Queries
-  module Projects
-    module Filters
-      class AncestorFilter < ::Queries::Projects::Filters::ProjectFilter
-        def apply_to(_query_scope)
-          case operator
-          when "="
-            super
-              .where(exists_condition.exists)
-          when "!"
-            super
-              .where.not(exists_condition.exists)
-          else
-            raise "unsupported operator"
-          end
-        end
+class Queries::Projects::Filters::AncestorFilter < Queries::Projects::Filters::Base
+  def apply_to(_query_scope)
+    case operator
+    when "="
+      super
+        .where(exists_condition.exists)
+    when "!"
+      super
+        .where.not(exists_condition.exists)
+    else
+      raise "unsupported operator"
+    end
+  end
```

Comment:
Out of curiosity, why/what changes when you use this syntax as compared to the former nested modules style?

Reply:
With the chained syntax all the modules need to already exist:

```ruby
>> class Foo::Bar::Baz; end
(irb):1:in `<main>': uninitialized constant Foo (NameError)

>> module Foo; module Bar; end; end;
>> class Foo::Bar::Baz; end
=> nil
```

Reply:
@aaron-contreras Main difference is constant resolution - in compact mode after current scope constant resolution will proceed right away to global scope, and in nested it will use every nested scope on the way

@klaustopher True in Ruby, but not when rails is involved - [zeitwerk handles implicit namespaces](https://github.com/fxn/zeitwerk?tab=readme-ov-file#implicit-namespaces)